src/core/public/keyboard_shortcuts/keyboard_shortcuts_service.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import { KeyboardShortcutsSetup, KeyboardShortcutsStart, ShortcutDefinition } from './types';
+
+/** @internal */
+export class KeyboardShortcutsService {
+  // Phase 1.6: Basic Registration System - Simple shortcut storage
+  private shortcuts = new Map<string, ShortcutDefinition>();
+  private isListening = false;
```

Comment:
do we need this? can we just start listening and stop listening in the start() and stop()?

Reply:
nvm i think its safe to have to avoid possible double event listeners

Reply:
nvm part 2: since its passing a reference when we start event listener, it will never be registered twice. There will always be one event listener no matter how many times you call `start`